logstash-core/src/main/java/org/logstash/ackedqueue/Queue.java

```diff
@@ -255,6 +264,7 @@ public long write(Queueable element) throws IOException {
 
             this.headPage.write(data, seqNum, this.checkpointMaxWrites);
             this.unreadCount++;
+            this.currentSize += data.length;
```

Comment:
- when a page is created, is already uses all the disk space, writing to it does not increase its disk size. 
- because we block after write, we have to assume there is always space for 1 more write which  can mean creating a new page. 
- so either we accept that the max + 1 page can happen or we block after the first write to a new page if that page has reached the threshold even if it means that the page will be mostly "empty" in term of content (but not disk used, as I said an empty page is already pre-allocated on disk to its full capacity) because we cannot know if a 2nd element will fit in or require another page creation.